frameit/lib/frameit/editor.rb

```diff
@@ -347,10 +350,64 @@ def build_title_images(max_width, max_height)
           i.interline_spacing interline_spacing if interline_spacing
           i.fill fetch_config[key.to_s]['color']
         end
-        title_image.trim # remove white space
+
+        # Trimming the image will result in the loss of the common baseline between the text in all images.
+        # title_image.trim # remove white space
 
         results[key] = title_image
+
+        # Hence retrieve the calculated trim bounding box:
+        # CALCULATED: trim bounding box (without actually trimming), see: http://www.imagemagick.org/script/escape.php
+        calculated_trim_box = title_image.identify { |b| b.format("%@") }
+
+        # Get vertical top offset from the trim box by converting it from string to an integer array:
+        trim_values = calculated_trim_box.split(/[\sx+]/).map(&:to_i) # The format is "wxh+X+Y", so use multiple string separators: the whitespace "\s", "x" and "+"
+
+        # Get the minimum top offset of the trim box:
+        if trim_values[3] < top_vertical_trim_offset
+          top_vertical_trim_offset = trim_values[3]
+        end
+
+        # Get the maximum bottom offset of the trimbox, this is the top offset + height:
+        if (trim_values[3] + trim_values[1]) > bottom_vertical_trim_offset
+          bottom_vertical_trim_offset = trim_values[3] + trim_values[1]
+        end
+
+        # Store for the crop action:
+        trim_boxes[key] = trim_values
+      end
+
+      # Crop images based on top_vertical_trim_offset to maintain text baseline:
+      words.each do |key|
+        # Get matching trim box:
+        trim_box = trim_boxes[key]
+
+        # Determine the trim area by maintaining the same vertical top offset based on the smallest value from all trim boxes (top_vertical_trim_offset).
+        # The 4th parameter is the vertical top offset. When it's larger than the smallest vertical top offset, the trim box needs to be adjusted:
+        if trim_box[3] > top_vertical_trim_offset
+          # Increase the height of the trim box (2nd parameter) with the difference in vertical top offset:
+          trim_box[1] += trim_box[3] - top_vertical_trim_offset
+          # Change the vertical top offset to match that of the others:
+          trim_box[3] = top_vertical_trim_offset
```

Comment:
it seems like we use index 1 and 3 quite a bit, could we stick them into their own, more descriptive variables?

Reply:
See previous comment

Reply:
Although..... Adding these 4 parameters would trigger the 320 lines Class limit again.
Or is there an overall Class with constants in the project?

Reply:
Lets not worry about the 320 lines for now. Lets make the code the best we can and we'll worry about trimming it down on our end.

Reply:
Do you mind if I restore the 1 line that you changed earlier to keep within the 320 limit? We both agreed that didn't result in the best code.

Reply:
Thats cool! I plan to go in and make some changes before merging to keep the lines down. 😄

Reply:
Can I just commit my changes simultaneously?

Reply:
Sure!